### PR TITLE
Twilio

### PR DESCRIPTION
**Pull request by dillonhammer**
# Purpose

Rework notifications to use Twilio

<br>

# Tickets

https://trello.com/c/CxbutQVM/209-move-our-notification-system-away-from-facebook-notifications-to-twilio

# Contributors

@dillonhammer
@megandouli 

# Feature List

- Add a server on port 8080 listening for twilio-related requests
   - optional in dev environments; there's a new `dev:notifs` command to run the notification server in dev mode
- Create a signed JWT on successful Twilio phone # verification, check the JWT on every request to add/remove subscriptions
- Add new tables to database to keep track of users and followed courses/sections

# Notes

New environment vars:
```
TWILIO_PHONE_NUMBER
TWILIO_ACCOUNT_SID
TWILIO_AUTH_TOKEN
TWILIO_VERIFY_SERVICE_ID
CLIENT_ORIGIN
JWT_SECRET
```

<br>

# Checklist

- [x] Filled out PR template :wink:
- [ ] Approved by designers
- [x] Is passing linting checks
- [x] Is passing tsc
- [x] Is passing existing tests
- [x] Has documentation and comments in code
- [x] Has test coverage for code
- [x] Will have clear squash commit message

<br>
@sandboxnu/searchneu

### CODE REVIEW

**Review by andrewydai — CHANGES_REQUESTED** [review]
A general comment is i feel we could have more comments explaining how pieces of code work. However, i'm also not sure how strict we are with that (for example class comments). Ideally I prefer the "Code is self explanatory" style but it's not always possible with complex code.

File: twilio/server.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+app.use(cors(corsOptions));
+const port = 8080;
+app.use(express.json());
+const server = createServer(app);
+
+server.listen(port, () => {
+  console.log("Running twilio notification server on port %s", port);
+});
+
+app.get("/knockknock", (req, res) => res.send("Who's there?"));
```
Comment (dfarooq-chwy) [inline twilio/server.ts:23]: I was bored and curious so I was poking around for a bit. This actually made me chuckle, thank you 😄

Comment (dfarooq610) [inline twilio/server.ts:23]: I was bored and curious so i am poking around. This actually made me chuckle 😄

File: services/notifyer.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    if (section.seatsRemaining > 0) {
+      message = `A seat opened up in ${section.subject} ${section.courseId} (CRN: ${section.crn}). Check it out at https://searchneu.com/${section.campus}/${section.termId}/search/${section.subject}${section.courseId} !`;
+    } else {
+      message = `A waitlist seat has opened up in ${section.subject} ${section.courseId} (CRN: ${section.crn}). Check it out at https://searchneu.com/${section.campus}/${section.termId}/search/${section.subject}${section.courseId} !`;
+    }
+    return message;
+  }
+}
+
+const instance = new Notifyer();
```
[the remark this answers is not in the document]
Comment (andrewydai) [reply services/notifyer.ts:86]: I read about this later and I now understand this is a common singleton pattern in TS/JS. I'm still not sure how i feel about it's necessity if there's no properties on the class but up to you

**Review by andrewydai — APPROVED** [review]
Thanks for all the hard work!

**Pull request merged**